### ISSUE TITLE
Modularize JavaScript Files

### ISSUE THREAD

**Issue by iansteyn** [EricHarrison72/gitGoblins#144]
Attempt to modularize our JavaScript code so that we don't have to re-write code, and so that code with similar functionality is consistent.

**Comment by iansteyn** [comment]
# I Give Up
I have spent so long and gotten nowhere. At this point I have sunk so many hours into what should be an extremely simple problem that I am giving up.

## The Goal
I wanted to separate our JavaScript code into a few different modules to make it more readable and re-usable, as I found we were re-writing similar functions across different files.

## The Problem
### JS Module Formats
Importing/Exporting in JS isn't quite as simple as in Python or Java. Because of JavaScript's history, there are many different formats for different frameworks. Since we are already using some NodeJS stuff, that's what I focused on. The two main formats are:
- **CommonJS modules** - this is the format Eric used to export functions to the JS testing file for the map stuff. It used to be NodeJS's only supported format, and is still the default.
- **ES6 modules** - a newer format that is the suggested standard for modern scripting languages, and is supposedly supported by modern browsers. NodeJS provides support for this too.

### Struggles :(
I tried so many things. Different formats, configuring `package.json`, using editor and browser debugging tools, changing the file extensions. I read articles, I searched forums, I watched videos. But in the end there are two big roadblocks:
1. _**We are not building a NodeJS app.**_ Almost all of the resources online focus on building a JS focused application, which is not the case for us. We only need some minor JS functionality to support other tools, and this means that our JS files are not set up like a normal Node.js application.
2. _**Imports/Exports aren't working on browser-side.**_ Eric's exports for testing worked just fine. But it seems that trying to use module patterns for when the app is actually live is not so easy.

## Potential Solutions
1. _**Using a browser JS compiler like [Browserify](https://github.com/browserify/browserify-handbook?).**_ This lets you write separate NodeJS modules and then compiles them into a single browser-friendly file. 
2. _**Hosting our own web server.**_ The problem could also be with the fact that we are currently running the app on a local development server. It seems that this modular stuff sometimes works better if you are using an actual web server.
3. _**Strip the JS**_. Refactor our entire project to use as little JS as possible. Use Python modularity.

### Why we aren't using these solutions
The solutions above require either huge reworks, spending significant amounts of time learning how to use and install more external tools, or both. And the thing is - we don't need it that badly. With the limited scope of our project, and the fact that it will likely never actually be deployed, I think we can get away with some less-than-ideal design patterns in the small JS portion of the app.

# TL;DR
Modularizing our JS for the sake of good design is a problem that requires more time and energy than it is worth right now. We're gonna have to settle for re-writing a few functions. I will focus on reworking the html templates to ensure I'm not re-writing _everything_ Josh and Eric did.
